source/nordic_sdk/components/drivers_nrf/delay/nrf_delay.h

```diff
@@ -111,7 +111,7 @@ loop
         BX     LR
 }
 
-#elif defined ( __ICCARM__ )
+#elif defined ( __ICCARM__ ) || (__ARMCC_VERSION)
```

Comment:
Is this valid for both armcc5 and armcc6?
Is the macro `__ARMCC_VERSION` is defined in both both armcc5 and armcc6?

Reply:
Good point, I think I may actually need to check __ARMCC_VERSION is greater or equal than 6

Reply:
@geowor01 has pointed out that as it's an elif it's not strictly necessary to check

I'm probably going to add the check though to improve readability? Any thoughts either way?

Reply:
Readability is king.
However, if `defined ( __ICCARM__ ) || (__ARMCC_VERSION)` works to check if both word are defined, I'm honestly not 100% sure what `defined ( __ICCARM__ ) || (__ARMCC_VERSION>=600)` does exactly.

Maybe `#if (__ARMCC_VERSION >= 6...) || defined( __ICCARM__ )`?

Reply:
Actually is fine, I got a bit confused thinking that `define` was somehow working after the `||`.
in reality it was just checking first if __ICCARM__ was defined, and then if __ICCARM__ had a "truthy" value. So `defined ( __ICCARM__ ) || (__ARMCC_VERSION >= 6000000)` is of course fine.